source/aws_profile.c

```diff
@@ -30,6 +30,11 @@ struct aws_profile {
     bool has_profile_prefix;
 };
 
+struct aws_section_collection {
+    struct aws_allocator *allocator;
+    struct aws_hash_table sections;
+};
+
 struct aws_profile_collection {
     struct aws_allocator *allocator;
     enum aws_profile_source_type profile_source;
```

Comment:
I'm making comments about the line below this: struct aws_hash_table profiles; 

1) Change the variable name from ~profiles~ to `sections`. Leaving the name unchanged puts the code into a very confusing state where sometimes "profile" means "profile or session". I get that we can't change the public API, but we can make the internal stuff more understandable going forward.

2) Add a comment documenting what is IN this hashtable. It sucks that our C datastructures are essentially untyped, so comments really help.
```
    /* Map from "profile" or "sso-session" to ???
     * key: struct aws_string*
     * value: struct ???
     */
    struct aws_hash_table sections_by_type
```

OR just have two different hash-tables, one for profiles, one for sso-sessions. It doesn't scale, but it makes sense for now

Reply:
do you really need another hash table? why cant top level just be an array? have a static var 0 refer to profiles, 1 to session tokens and so on for other section types?

Reply:
having it as array is arguable: if we want api to always deal with concrete session type (i.e. profiles, session token, etc...) then static array is fine. if we want to support parsing of generic section types then we hash table is probably better

Reply:
Thanks, I have updated it to an Array+Enum which is much better in terms of clarity.